src/SDK/MonitoringApplianceEmulator/Models/EmulationRunSettings.cs

```diff
@@ -112,19 +112,22 @@ public static EmulationRunSettings LoadEmulationRunSettings(string filePath)
         /// <summary>
         /// Saves the emulation run settings to a file.
         /// </summary>
-        public void Save()
+        /// <param name="detectorName">The detector name.</param>
+        public void Save(string detectorName)
         {
             try
             {
                 string emulatorRunSettingsPath = Path.Combine(EmulationAlertsFilePath, $"{this.SubscriptionId}");
 
-                // Creates a folder for saving the emulation run settings
+                // Create a folder for saving the emulation run settings
                 Directory.CreateDirectory(emulatorRunSettingsPath);
 
-                // Creates the path of the file with the emulation run settings
-                string filePath = Path.Combine(emulatorRunSettingsPath, $"{this.StartTime:yyyy-MM-dd HH-mm-ss}_{this.EndTime:yyyy-MM-dd HH-mm-ss}_{this.AnalysisCadence.TimeSpan:hh'-'mm'-'ss}_{this.EmulationAlerts.Count}.json");
+                // Create file path with the emulation run settings
+                string filePath = (this.IterativeRunModeEnabled) ?
+                    Path.Combine(emulatorRunSettingsPath, $"{detectorName}_{this.StartTime:yyyy-MM-dd HH-mm-ss}_{this.EndTime:yyyy-MM-dd HH-mm-ss}_{this.AnalysisCadence.TimeSpan:hh'-'mm'-'ss}.json") :
+                    Path.Combine(emulatorRunSettingsPath, $"{detectorName}_{this.StartTime:yyyy-MM-dd HH-mm-ss}.json");
 
-                using (StreamWriter writer = File.CreateText(filePath))
```

Comment:
>CreateText [](start = 50, length = 10)

why did you need to change it?

Reply:
I changed it back to how it was before.

---
In reply to: [238683625](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/32#discussion_r238683625) [](ancestors = 238683625,238670129)